test/asymmetric_test.dart

```diff
@@ -0,0 +1,117 @@
+////////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2023 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////////
+
+import 'package:test/expect.dart' hide throws;
+
+import '../lib/realm.dart';
+import 'test.dart';
+
+part 'asymmetric_test.g.dart';
+
+@RealmModel(ObjectType.asymmetricObject)
+class _Asymmetric {
+  @PrimaryKey()
+  @MapTo('_id')
+  late ObjectId id;
+
+  late List<_Embedded> embeddedObjects;
+}
+
+@RealmModel(ObjectType.embeddedObject)
+class _Embedded {
+  late int value;
+  late RealmValue any;
+  _Symmetric? symmetric;
+}
+
+@RealmModel()
+class _Symmetric {
+  @PrimaryKey()
+  @MapTo('_id')
+  late ObjectId id;
+}
+
+Future<void> main([List<String>? args]) async {
+  await setupTests(args);
+
+  Future<Realm> getSyncRealm(AppConfiguration config) async {
```

Comment:
Is it allowed to use Asymmetric schema in a local realm configuration? If not, do we have a validation?

Reply:
No it is not, and yes

Reply:
Added a test

Reply:
I don't see it, probably it is in core. If you don't mind, you can add some simple test about it.